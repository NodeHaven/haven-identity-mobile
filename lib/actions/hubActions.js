// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import { STORE_HUB_HEAD, CATCHUP_HUB, QUEUE_EVENT, TAKE_QUEUED_EVENT, CLEAR_QUEUE, IMPORT_SNAPSHOT, RESET_HUB } from '../constants/HubActionTypes'

export function storeHubHead (hash) {
  return {
    type: STORE_HUB_HEAD,
    hash
  }
}
export function resetHub () {
  return {
    type: RESET_HUB
  }
}

export function catchupHub () {
  return {
    type: CATCHUP_HUB
  }
}

export function queueEvent (event) {
  return {
    type: QUEUE_EVENT,
    event
  }
}

export function takeQueuedEvent () {
  return {
    type: TAKE_QUEUED_EVENT
  }
}

export function clearQueue () {
  return {
    type: CLEAR_QUEUE
  }
}

export function importSnapshot (snapshot) {
  return {
    type: IMPORT_SNAPSHOT,
    snapshot
  }
}
