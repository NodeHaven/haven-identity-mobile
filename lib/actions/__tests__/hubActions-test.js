// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//

import { storeHubHead, catchupHub, queueEvent, takeQueuedEvent, clearQueue, importSnapshot } from 'uPortMobile/lib/actions/hubActions.js'

it('creates a STORE_HUB_HEAD action', () => {
  expect(storeHubHead('abcdefg')).toMatchSnapshot()
})

it('creates a CATCHUP_HUB action', () => {
  expect(catchupHub()).toMatchSnapshot()
})

it('creates a STORE_HUB_HEAD action', () => {
  expect(queueEvent({type: 'SOME_EVENT', data: 'stuff'})).toMatchSnapshot()
})

it('creates a CATCHUP_HUB action', () => {
  expect(takeQueuedEvent()).toMatchSnapshot()
})

it('creates a CLEAR_QUEUE action', () => {
  expect(clearQueue()).toMatchSnapshot()
})

it('creates an IMPORT_SNAPSHOT action', () => {
  expect(importSnapshot({uport: [], hdwallet: {}})).toMatchSnapshot()
})
