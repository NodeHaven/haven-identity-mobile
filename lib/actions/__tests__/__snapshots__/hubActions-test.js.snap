// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a CATCHUP_HUB action 1`] = `
Object {
  "type": "CATCHUP_HUB",
}
`;

exports[`creates a CATCHUP_HUB action 2`] = `
Object {
  "type": "TAKE_QUEUED_EVENT",
}
`;

exports[`creates a CLEAR_QUEUE action 1`] = `
Object {
  "type": "CLEAR_QUEUE",
}
`;

exports[`creates a STORE_HUB_HEAD action 1`] = `
Object {
  "hash": "abcdefg",
  "type": "STORE_HUB_HEAD",
}
`;

exports[`creates a STORE_HUB_HEAD action 2`] = `
Object {
  "event": Object {
    "data": "stuff",
    "type": "SOME_EVENT",
  },
  "type": "QUEUE_EVENT",
}
`;

exports[`creates an IMPORT_SNAPSHOT action 1`] = `
Object {
  "snapshot": Object {
    "hdwallet": Object {},
    "uport": Array [],
  },
  "type": "IMPORT_SNAPSHOT",
}
`;
